api/src/app.ts

```diff
@@ -222,14 +222,18 @@ export const build = async (
       await fastify.register(settingRedirectRoutes);
     });
   });
-  // Routes not requiring authentication:
-  void fastify.register(mobileAuth0Routes);
-  // TODO: consolidate with LOCAL_MOCK_AUTH
-  if (FCC_ENABLE_DEV_LOGIN_MODE) {
-    void fastify.register(devAuthRoutes);
-  } else {
-    void fastify.register(authRoutes);
-  }
+  // Routes for signed out users:
+  void fastify.register(async function (fastify) {
+    fastify.addHook('onRequest', fastify.authorize);
+    fastify.addHook('onRequest', fastify.redirectIfSignedIn);
```

Comment:
This seems contradictory: Register `redirectIfSignedIn`, if user is signed out.

Reply:
Also, correct me if I am wrong, but the old api does not have `redirectIfSignedIn` for `/auth/auth0/callback`, but this adds it to the new api.

I am not sure when that would happen, but it does seem like a change.

Reply:
Is this contradictory? The idea was these routes are for signed out users only, so if you're signed in, then it should send you away.

> old api does not have redirectIfSignedIn for /auth/auth0/callback, but this adds it to the new api.

Fair point, I'll make it more focused and add TODOs to go back to this once we're confident that nothing breaks.

Reply:
> The idea was these routes are for signed out users only, so if you're signed in, then it should send you away.

What route is available to non-signed in users, but not signed-in users? Just `/signin`?

Reply:
Also `mobile-login`. I just wanted to add the plugin first.

Reply:
Yeah, looking at this again, I am not sure we should have cared about something like `/auth/...`